client/src/pages/ProfileList/profileList.tsx

```diff
@@ -1,23 +1,84 @@
-import { Button, TextField, Grid } from '@mui/material';
+import { Button } from '@mui/material';
 import { Box } from '@mui/system';
-import InputAdornment from '@mui/material/InputAdornment';
-import SearchIcon from '@mui/icons-material/Search';
+import React, { useState, useEffect } from 'react';
+import { useSnackBar } from '../../context/useSnackbarContext';
 import useStyle from './useStyles';
+import { Profile } from '../../interface/ProfileApiData';
 import Typography from '@mui/material/Typography';
+import { Grid } from '@mui/material';
 import ProfileCard from './profileCard/ProfileCard';
+import { getAllSitter } from '../../helpers/APICalls/getSitter';
 import SearchBar from './searchBar';
+
 function ProfileList(): JSX.Element {
+  const { updateSnackBarMessage } = useSnackBar();
   const classes = useStyle();
-  const cards = [1, 2, 3, 4, 5, 6];
-
+  const [location, setLocation] = useState<string>('');
+  const [profiles, setProfiles] = useState<Profile[]>([]);
+  const [filterprofiles, setFilterProfiles] = useState<Profile[]>([]);
+  const [addMore, setAddMore] = useState<Profile[]>([]);
+  const [disable, setDisable] = useState(true);
+  const [noRecord, setNoRecord] = useState('');
+  useEffect(() => {
+    setAddMore([]);
+  }, [location]);
+  const saveProfiles = async (profilesdata: Profile[]) => {
+    setProfiles([]);
+    setFilterProfiles([]);
+    addMore.length = 0;
+    const length = profilesdata.length;
+    setProfiles([...profilesdata]);
+    setFilterProfiles([...profilesdata]);
+    const data = profilesdata.splice(0, 6);
+    data.map((ele) => {
+      addMore.push(ele);
+    });
+    setAddMore([...addMore]);
+    setDisable(false);
+    if (length > addMore.length) {
+      const btn = document.getElementById('showMore') as HTMLElement;
+      btn.style.display = 'block';
+    }
+  };
+  const changeLocation = (e: any) => {
+    setLocation(e.target.value);
+    const btn = document.getElementById('showMore') as HTMLElement;
+    btn.style.display = 'none';
+  };
+  const handleShowMore = () => {
```

Comment:
Ideally, we want to paginate on the server side - which is a seperate ticket. The current implementation seems a bit over engineered but I guess we could leave it in for now :)